### DISCUSSION TITLE
Bash completions for Homebrew does not correctly work @ Mac

### DISCUSSION THREAD

**Discussion by y-vectorfield** [Homebrew/discussions#4322] [Everyday usage]
### Output of `brew config`

```shell
$ brew config
HOMEBREW_VERSION: 4.0.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0c1031f2fd71016f0f5df699caaed384487bfb16
Last commit: 4 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a0c6d795116fcc928b9e326150944750d944a013
Core tap last commit: 3 weeks ago
Core tap branch: master
Core tap JSON: 14 Mar 07:08 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.3-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
$ brew doctor
Your system is ready to brew.
```

### Description of issue

Bash completions for homebrew does not correctly work.

brew do(Press Tab key)
brew Documents/
(Desired result: brew doctor)

Of course, I set the settings for completion in .bash_profile which is explained following link.
https://docs.brew.sh/Shell-Completion

If someone had met this problem and could solve this, please give me your advice.

**Comment by y-vectorfield** [comment]
I got a answer of this problem.

echo "HOMEBREW_REPOSITORY=$(brew --repository)" >> .bash_profile

**Reply by gromgit** [reply]
The Homebrew installer would have told you to add:
```
eval "$(/path/to/brew shellenv)"
```
to your `.bash_profile`, which correctly sets `HOMEBREW_REPOSITORY` and other important variables like `PATH`.

**Comment by y-vectorfield** [comment]
@gromgit 
Thank you for your comment.
I updated Homebrew, but above setting did not add my .bash_profile.

**Reply by gromgit** [reply]
I was talking about the official Homebrew installer, i.e.:
```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
which you should only run once...when you install Homebrew itself. That installer prints a **Next Steps** list of actions when it finishes, and the command I mentioned earlier is one of those actions.

For your current situation, I suggest you do the following:
1. Run `brew shellenv` manually to see exactly what it sets.
2. For each of your shell startup files, replace _all_ the `HOMEBREW_*`=... lines that you saw in [1] with the single command `eval "$(/path/to/brew shellenv)"`.
3. Log out, then log in again, then test your `brew` completions again.

**Reply by y-vectorfield** [reply]
@gromgit 
Thank you very much for your comment.
I updated homebrew 4.0 using brew upgrade, so I might forget to see Next Steps.
I manually added above setting in .bash_profile and ran brew, the completions correctly worked!!
